src/test/ReactTestUtils.js

```diff
@@ -309,6 +309,44 @@ var ReactTestUtils = {
     return new ReactShallowRenderer();
   },
 
+  jasmineMatchers: {
+    /**
+    * Expect function to perform a console.warn.
+    * If `expectation` is a number, we expect the function to make `expectation` calls to warn.
+    * If `expectation` is a string, we expect that string to appear in the warn output.
+    **/
+    toWarn: function(expectation) {
+      mocks = require('mocks');
+      var warn = console.warn;
+      console.warn = mocks.getMockFunction();
```

Comment:
use `spyOn(console, 'warn')` (andCallThrough maybe), not the mocks module. That gets rid of the try/finally because we don't need to reset console.